### PR TITLE
timesync doc

### LINKED ISSUE #6782: design detail of timesync
please update the design of timesync

### PR DESCRIPTION
**Pull request by neza2017**
update design detail of timesync
Resolves: #6782 
Signed-off-by: yefu.chen <yefu.chen@zilliz.com>

### CODE REVIEW

File: docs/design_docs/milvus_timesync_en.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+1. If `u1` and `u2` are on different nodes, and their time is not synchronized. To give an extreme example, suppose that the time of `u2` is 24 hours later than `u1`, then all the operations of `u1` can't been seen by `u2` until next day.
+2. Network latency. If `u2` starts the `Search on C0` at `t17`, then how to ensure that all the `events` before `t17` have been processed. If the envents of `delete A1 from C0` has been delayed dure to the network latency, then it would lead to incorrect state: `u2` would see both `A1` and `A2` at `t17`.
+
+`Time synchronization system` is used to solve the above problems.
+
+## Timestamp Oracle(TSO)
+
+Like [TiKV](https://github.com/tikv/tikv), `Milvus 2.0` provides `TSO` service, all the events must alloc timestamp from `TSO`，not use local timestamp any more, so the first problem should be solved.
```
Comment (xiaofan-luan) [inline docs/design_docs/milvus_timesync_en.md:28]: TSO is a very general solution and it's not invented by tikv.
If we want to talk about timestamp strategy, we may want to mention tidb, cockroach and spanner also

File: docs/design_docs/milvus_timesync_en.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    uint64 timestamp = 2;
+    uint32 count = 3;
+}
+
+```
+`Timestamp` is type of `uint64`, contains physical and logical parts. 
+
+This is the format of `Timestamp`
+
+![Timestamp struct](./graphs/time_stamp_struct.png)
```
Comment (xiaofan-luan) [inline docs/design_docs/milvus_timesync_en.md:57]: this is weird, why is it 48 + 18?

Comment (neza2017) [reply docs/design_docs/milvus_timesync_en.md:57]: We did not do any deep research on the struct of timestamp,but tikv' timestamp use this struct, so we think it should also works for milvus.

File: docs/design_docs/milvus_timesync_en.md (hunk 1 above)
Comment (neza2017) [reply docs/design_docs/milvus_timesync_en.md:28]: File 'internal/tso/tso.go' is copied from 'TiKV', so I mentioned 'TiKV' here

**Pull request merged**